shared/strbuf.c

```diff
@@ -39,15 +39,17 @@ static bool buf_realloc(struct strbuf *buf, size_t sz)
 	return true;
 }
 
-static bool buf_grow(struct strbuf *buf, size_t newsize)
+bool strbuf_reserve(struct strbuf *buf, size_t n)
```

Comment:
Would be great if we can have a way to instrument the compiler/analyzers that follow-up `strbuf_str()` and `strbuf_push*()` calls cannot fail, when reservation large enough. As humans we'll likely get it wrong at some point.

Any clever trick that comes to mind?

Reply:
I think it's worthy. In several places we have the strategy "1) calculate the size; 2) allocate; 3) use it". Allowing to reserve space is IMO a good thing to do

Reply:
^^ :100: